exporter/kafkaexporter/raw_marshaler.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafkaexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/kafkaexporter"
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/Shopify/sarama"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+var errUnsupported = errors.New("unsupported serialization")
+
+type rawMarshaler struct {
+}
+
+func newRawMarshaler() rawMarshaler {
+	return rawMarshaler{}
+}
+
+func (r rawMarshaler) Marshal(logs plog.Logs, topic string) ([]*sarama.ProducerMessage, error) {
+	messages := make([]*sarama.ProducerMessage, logs.LogRecordCount())
+
+	index := 0
+	for i := 0; i < logs.ResourceLogs().Len(); i++ {
+		rl := logs.ResourceLogs().At(i)
+		for j := 0; j < rl.ScopeLogs().Len(); j++ {
+			sl := rl.ScopeLogs().At(j)
+			for k := 0; k < sl.LogRecords().Len(); k++ {
+				lr := sl.LogRecords().At(k)
+				b, err := r.logBodyAsBytes(lr.Body())
+				if err != nil {
+					return nil, err
+				}
+
+				messages[index] = &sarama.ProducerMessage{
```

Comment:
I suggest that we avoid sending an empty buffer to avoid that overhead on kafka's side.

Reply:
OK fixed